File: problems/0007_reverse-integer/Solution.java
class Solution {
    public int reverse(int x) {
        final int sign = Integer.signum(x);
        x *= sign;
        long result = 0;
        for (; x != 0; x /= 10)
            result = result * 10 + (x % 10);
        result *= sign;
        return (result < Integer.MIN_VALUE || Integer.MAX_VALUE < result ?
               0 : (int) result);
    }
}

File: problems/0007_reverse-integer/Solution.kt
class Solution {
    fun reverse(x: Int): Int {
        var n = x
        var result = 0L
        while (n != 0) {
            result = result * 10 + (n % 10)
            n /= 10
        }
        return if (result < Int.MIN_VALUE || Int.MAX_VALUE < result)
                    0
                else
                    result.toInt()
    }
}
